File: backend/app/schemas/video.py
"""视频项目相关 Pydantic 模型"""
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class VideoProjectCreate(BaseModel):
    title: str
    script_id: Optional[int] = None
    voice_id: Optional[int] = None
    bgm_path: Optional[str] = None
    bgm_volume: float = 0.3
    material_paths_json: str = "[]"
    aspect_ratio: str = "9:16"
    subtitle_enabled: int = 1
    lip_sync_enabled: int = 0
    lip_sync_mode: str = "pip"
    image_animation_type: Optional[str] = None
    subtitle_animation: str = "fade"
    collected: int = 0


class VideoProjectUpdate(BaseModel):
    title: Optional[str] = None
    bgm_path: Optional[str] = None
    bgm_volume: Optional[float] = None
    material_paths_json: Optional[str] = None
    subtitle_enabled: Optional[int] = None
    lip_sync_enabled: Optional[int] = None
    lip_sync_mode: Optional[str] = None
    image_animation_type: Optional[str] = None
    subtitle_animation: str = "fade"
    collected: int = 0


class VideoProjectOut(BaseModel):
    id: int
    title: str
    script_id: Optional[int] = None
    voice_id: Optional[int] = None
    bgm_path: Optional[str] = None
    bgm_volume: float
    material_paths_json: str
    subtitle_enabled: int
    lip_sync_enabled: int = 0
    lip_sync_mode: str = "pip"
    image_animation_type: Optional[str] = None
    subtitle_animation: str = "fade"
    collected: int = 0
    aspect_ratio: str = "9:16"
    status: str
    output_path: Optional[str] = None
    duration_seconds: Optional[float] = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/video.ts
export interface VideoProject {
  id: number
  title: string
  script_id?: number
  voice_id?: number
  bgm_path?: string
  bgm_volume: number
  material_paths_json: string
  subtitle_enabled: number
  lip_sync_enabled: number
  lip_sync_mode: string
  image_animation_type?: string
  subtitle_animation: string
  aspect_ratio: string
  status: string
  collected: number
  output_path?: string
  duration_seconds?: number
  created_at: string
  updated_at: string
}

export interface AsyncTask {
  id: number
  task_type: string
  ref_id?: number
  celery_task_id?: string
  status: string
  progress: number
  progress_message?: string
  error_message?: string
  result_json: string
  created_at: string
  updated_at: string
}
